storage/src/stream.cc

```diff
@@ -555,26 +556,29 @@ unsigned int stream::_find_index_id(
         << service_id << ") (host: " << host_name << ", service: "
         << service_desc << ", special: " << special << ")";
       // Update index_data table.
-      std::ostringstream query;
-      query << "UPDATE " << (db_v2 ? "index_data" : "rt_index_data")
-            << "  SET host_name=:host_name,"
-               "     service_description=:service_description,"
-               "     special=:special"
-               "  WHERE host_id=:host_id"
-               "    AND service_id=:service_id";
+      if (!_index_data_update.prepared()) {
+        std::ostringstream query;
+        query << "UPDATE index_data"
+                 " SET host_name=:host_name,"
+                 " service_description=:service_description,"
+                 " special=:special"
+                 " WHERE host_id=:host_id"
+                 " AND service_id=:service_id";
+        _index_data_update.prepare(query.str());
+      }
       try {
-        database_query q(_db);
-        q.prepare(query.str());
-        q.bind_value(":host_name", host_name);
-        q.bind_value(":service_description", service_desc);
-        q.bind_value(":special", special);
-        q.bind_value(":host_id", host_id);
-        q.bind_value(":service_id", service_id);
-        q.run_statement();
+        _index_data_update.bind_value(":host_name", host_name);
+        _index_data_update.bind_value(":service_description", service_desc);
+        QString sp(QString::number(special));
+        _index_data_update.bind_value(":special", sp);
```

Comment:
If I remember well, special is 0 or 1, so you could avoid the QString which is an heavy object.
`_index_data_update.bind_value(":special", sp == 0 ? "0" : "1");`
or a condition if you prefer would be better.